prov/shm/src/smr_atomic.c

```diff
@@ -232,31 +178,47 @@ static ssize_t smr_generic_atomic(struct smr_ep *ep,
 		assert(result_ioc);
 		ofi_ioc_to_iov(result_ioc, result_iov, result_count,
 			       ofi_datatype_size(datatype));
-		if (!domain->fast_rma)
-			flags |= SMR_RMA_REQ;
+		flags |= SMR_RMA_REQ;
 		/* fall through */
 	case ofi_op_atomic:
 		if (atomic_op != FI_ATOMIC_READ) {
 			assert(ioc);
 			ofi_ioc_to_iov(ioc, iov, count, ofi_datatype_size(datatype));
 		} else {
+			flags |= SMR_RMA_REQ;
```

Comment:
Going to remove this in the next revision, but holding off to not overload the CI.
Atomic reads should always go through the fetch case so it should already be set. All ops fall through to here also so this is getting set for all ops, incorrectly.